go/plugins/vertexai/vertexai.go

```diff
@@ -16,15 +16,97 @@ package vertexai
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"runtime"
 
+	aiplatform "cloud.google.com/go/aiplatform/apiv1"
 	"cloud.google.com/go/vertexai/genai"
 	"github.com/firebase/genkit/go/ai"
 	"github.com/firebase/genkit/go/plugins/internal/uri"
+	"google.golang.org/api/option"
 )
 
-func newClient(ctx context.Context, projectID, location string) (*genai.Client, error) {
-	return genai.NewClient(ctx, projectID, location)
+const provider = "google-genai"
+
+// Config configures the plugin.
+type Config struct {
+	// The project holding the resources.
+	ProjectID string
+	// The location of the resources.
+	// Defaults to "us-central1".
+	Location string
+	// Generative models to provide.
+	Models []string
+	// Embedding models to provide.
+	Embedders []string
+}
+
+func Init(ctx context.Context, cfg Config) (err error) {
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("vertexai.Init: %w", err)
+		}
+	}()
+
+	if cfg.ProjectID == "" {
+		return errors.New("missing ProjectID")
+	}
+	if cfg.Location == "" {
+		cfg.Location = "us-central1"
+	}
+	// TODO(#345): call ListModels. See googleai.go.
+	if len(cfg.Models) == 0 && len(cfg.Embedders) == 0 {
+		return errors.New("need at least one model or embedder")
+	}
+
+	// Client for Gemini SDK.
+	gclient, err := genai.NewClient(ctx, cfg.ProjectID, cfg.Location)
+	if err != nil {
+		return err
+	}
+
+	// Client for prediction SDK, needed for embedders.
```

Comment:
This code seems to alternate between model setup and embedder setup. Maybe we should do all one, then all of the other. Maybe in separate functions?

Reply:
Done.